anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/forecast/ForecastingDetectorTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.forecast;
+
+import com.expedia.adaptivealerting.anomdetect.forecast.interval.IntervalForecast;
+import com.expedia.adaptivealerting.anomdetect.forecast.interval.IntervalForecaster;
+import com.expedia.adaptivealerting.anomdetect.forecast.point.PointForecaster;
+import com.expedia.adaptivealerting.core.anomaly.AnomalyType;
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyDouble;
+import static org.mockito.Mockito.when;
+
+public class ForecastingDetectorTest {
+    private ForecastingDetector detectorUnderTest;
+
+    @Mock
+    private PointForecaster pointForecaster;
+
+    @Mock
+    private IntervalForecaster intervalForecaster;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        initDependencies();
+        this.detectorUnderTest = new ForecastingDetector(
+                UUID.randomUUID(),
+                pointForecaster,
+                intervalForecaster,
+                AnomalyType.TWO_TAILED);
+    }
+
+    @Test
+    public void testClassify() {
+        val metricDef = new MetricDefinition("my-metric");
```

Comment:
Suggest pulling "my-metric" out as a constant e.g. `DUMMY_METRIC_KEY = "dummy-metric"`.  
The "dummy" label can help demonstrate there's no magic going on behind the scenes and the test doesn't rely on the value.

Reply:
I redid this to pull the metric def from an object mother.

Reply:
I guess that doesn't really address the issue though. Maybe I can rename the variable to `dummyMetricDef`.

Reply:
Hm. The variables are test-level fields in a lot of cases. Not sure we want to have to name all these "dummy-this", "dummy-that". My suggestion would be that we pick an example and then show me what you have in mind, and let's see whether it's a general pattern we want to adopt and how we want to do so.

FWIW I do like the idea of making it clearer when something is a dummy object. So not necessarily against dummy-this etc.